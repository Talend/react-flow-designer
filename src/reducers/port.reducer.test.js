import { Map, OrderedMap } from 'immutable';

import portReducer from './port.reducer';
import { PortRecord, PositionRecord } from '../constants/flowdesigner.model';

describe('Check port reducer', () => {

	const initialState = new Map().set('ports', new OrderedMap()
		.set('id1', new PortRecord({
			id: 'id1',
			position: new PositionRecord({ x: 10, y: 10 }),
		}))
		.set('id2', new PortRecord({
			id: 'id2',
			position: new PositionRecord({ x: 10, y: 10 }),
		}))
		.set('id3', new PortRecord({
			id: 'id3',
			position: new PositionRecord({ x: 10, y: 10 }),
		})))
        .set('nodes', new Map().set('nodeId', new Map()));

	it('FLOWDESIGNER_PORT_ADD properly add the port to the port OrderedMap', () => {
		expect(portReducer(initialState, {
			type: 'FLOWDESIGNER_PORT_ADD',
			nodeId: 'nodeId',
			portId: 'portId',
			portType: 'portType',
			attr: { clicked: true },
		})).toEqual(new Map().set('ports', new OrderedMap().set('id1', new PortRecord({
			id: 'id1',
			position: new PositionRecord({ x: 10, y: 10 }),
		}))
		.set('id2', new PortRecord({
			id: 'id2',
			position: new PositionRecord({ x: 10, y: 10 }),
		}))
		.set('id3', new PortRecord({
			id: 'id3',
			position: new PositionRecord({ x: 10, y: 10 }),
		}))
		.set('portId', new PortRecord({
			id: 'portId',
			nodeId: 'nodeId',
			portType: 'portType',
			attr: new Map({ clicked: true }),
		})))
        .set('nodes', new Map().set('nodeId', new Map())));
	});
});
