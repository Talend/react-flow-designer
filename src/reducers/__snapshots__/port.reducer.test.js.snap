exports[`Check port reducer FLOWDESIGNER_PORT_ADD properly add the port to the port OrderedMap 1`] = `
Object {
  "childrens": Object {},
  "in": Object {},
  "links": Object {},
  "nodeTypes": Object {},
  "nodes": Object {
    "nodeId": Object {},
  },
  "out": Object {
    "nodeId": Object {
      "portId": Object {},
    },
  },
  "parents": Object {},
  "ports": Object {
    "id1": Object {
      "data": Object {
        "type": "test",
      },
      "graphicalAttributes": Object {
        "attr": "attr",
      },
      "id": "id1",
      "nodeId": undefined,
      "portType": undefined,
      "position": Object {
        "x": 10,
        "y": 10,
      },
    },
    "id2": Object {
      "data": Object {},
      "graphicalAttributes": Object {
        "type": undefined,
      },
      "id": "id2",
      "nodeId": undefined,
      "portType": undefined,
      "position": Object {
        "x": 10,
        "y": 10,
      },
    },
    "id3": Object {
      "data": Object {},
      "graphicalAttributes": Object {
        "type": undefined,
      },
      "id": "id3",
      "nodeId": undefined,
      "portType": undefined,
      "position": Object {
        "x": 10,
        "y": 10,
      },
    },
    "portId": Object {
      "data": Object {},
      "graphicalAttributes": Object {
        "type": "EMITTER",
      },
      "id": "portId",
      "nodeId": "nodeId",
      "portType": "portType",
      "position": undefined,
    },
  },
  "transform": Object {
    "k": 1,
    "x": 0,
    "y": 0,
  },
  "transformToApply": undefined,
}
`;

exports[`Check port reducer FLOWDESIGNER_PORT_ADDS to add multiple ports (portId1, portId2) to port collection 1`] = `
Object {
  "childrens": Object {},
  "in": Object {
    "nodeId": Object {
      "portId2": Object {},
    },
  },
  "links": Object {},
  "nodeTypes": Object {},
  "nodes": Object {
    "nodeId": Object {},
  },
  "out": Object {
    "nodeId": Object {
      "portId1": Object {},
    },
  },
  "parents": Object {},
  "ports": Object {
    "id1": Object {
      "data": Object {
        "type": "test",
      },
      "graphicalAttributes": Object {
        "attr": "attr",
      },
      "id": "id1",
      "nodeId": undefined,
      "portType": undefined,
      "position": Object {
        "x": 10,
        "y": 10,
      },
    },
    "id2": Object {
      "data": Object {},
      "graphicalAttributes": Object {
        "type": undefined,
      },
      "id": "id2",
      "nodeId": undefined,
      "portType": undefined,
      "position": Object {
        "x": 10,
        "y": 10,
      },
    },
    "id3": Object {
      "data": Object {},
      "graphicalAttributes": Object {
        "type": undefined,
      },
      "id": "id3",
      "nodeId": undefined,
      "portType": undefined,
      "position": Object {
        "x": 10,
        "y": 10,
      },
    },
    "portId1": Object {
      "data": Object {},
      "graphicalAttributes": Object {
        "type": "EMITTER",
      },
      "id": "portId1",
      "nodeId": "nodeId",
      "portType": "portType",
      "position": undefined,
    },
    "portId2": Object {
      "data": Object {},
      "graphicalAttributes": Object {
        "type": "SINK",
      },
      "id": "portId2",
      "nodeId": "nodeId",
      "portType": "portType",
      "position": undefined,
    },
  },
  "transform": Object {
    "k": 1,
    "x": 0,
    "y": 0,
  },
  "transformToApply": undefined,
}
`;

exports[`Check port reducer FLOWDESIGNER_PORT_REMOVE should remove port id1 from ports collection 1`] = `
Object {
  "childrens": Object {},
  "in": Object {},
  "links": Object {},
  "nodeTypes": Object {},
  "nodes": Object {
    "nodeId": Object {},
  },
  "out": Object {},
  "parents": Object {},
  "ports": Object {
    "id2": Object {
      "data": Object {},
      "graphicalAttributes": Object {
        "type": undefined,
      },
      "id": "id2",
      "nodeId": undefined,
      "portType": undefined,
      "position": Object {
        "x": 10,
        "y": 10,
      },
    },
    "id3": Object {
      "data": Object {},
      "graphicalAttributes": Object {
        "type": undefined,
      },
      "id": "id3",
      "nodeId": undefined,
      "portType": undefined,
      "position": Object {
        "x": 10,
        "y": 10,
      },
    },
  },
  "transform": Object {
    "k": 1,
    "x": 0,
    "y": 0,
  },
  "transformToApply": undefined,
}
`;

exports[`Check port reducer FLOWDESIGNER_PORT_REMOVE_DATA remove type on port id1 on port data map 1`] = `
Object {
  "childrens": Object {},
  "in": Object {},
  "links": Object {},
  "nodeTypes": Object {},
  "nodes": Object {
    "nodeId": Object {},
  },
  "out": Object {},
  "parents": Object {},
  "ports": Object {
    "id1": Object {
      "data": Object {},
      "graphicalAttributes": Object {
        "attr": "attr",
      },
      "id": "id1",
      "nodeId": undefined,
      "portType": undefined,
      "position": Object {
        "x": 10,
        "y": 10,
      },
    },
    "id2": Object {
      "data": Object {},
      "graphicalAttributes": Object {
        "type": undefined,
      },
      "id": "id2",
      "nodeId": undefined,
      "portType": undefined,
      "position": Object {
        "x": 10,
        "y": 10,
      },
    },
    "id3": Object {
      "data": Object {},
      "graphicalAttributes": Object {
        "type": undefined,
      },
      "id": "id3",
      "nodeId": undefined,
      "portType": undefined,
      "position": Object {
        "x": 10,
        "y": 10,
      },
    },
  },
  "transform": Object {
    "k": 1,
    "x": 0,
    "y": 0,
  },
  "transformToApply": undefined,
}
`;

exports[`Check port reducer FLOWDESIGNER_PORT_REMOVE_GRAPHICAL_ATTRIBUTES to remove attr on port id1 graphicalAttribute map 1`] = `
Object {
  "childrens": Object {},
  "in": Object {},
  "links": Object {},
  "nodeTypes": Object {},
  "nodes": Object {
    "nodeId": Object {},
  },
  "out": Object {},
  "parents": Object {},
  "ports": Object {
    "id1": Object {
      "data": Object {
        "type": "test",
      },
      "graphicalAttributes": Object {},
      "id": "id1",
      "nodeId": undefined,
      "portType": undefined,
      "position": Object {
        "x": 10,
        "y": 10,
      },
    },
    "id2": Object {
      "data": Object {},
      "graphicalAttributes": Object {
        "type": undefined,
      },
      "id": "id2",
      "nodeId": undefined,
      "portType": undefined,
      "position": Object {
        "x": 10,
        "y": 10,
      },
    },
    "id3": Object {
      "data": Object {},
      "graphicalAttributes": Object {
        "type": undefined,
      },
      "id": "id3",
      "nodeId": undefined,
      "portType": undefined,
      "position": Object {
        "x": 10,
        "y": 10,
      },
    },
  },
  "transform": Object {
    "k": 1,
    "x": 0,
    "y": 0,
  },
  "transformToApply": undefined,
}
`;

exports[`Check port reducer FLOWDESIGNER_PORT_SET_DATA to merge { type: 'string' } on port id1 data map 1`] = `
Object {
  "childrens": Object {},
  "in": Object {},
  "links": Object {},
  "nodeTypes": Object {},
  "nodes": Object {
    "nodeId": Object {},
  },
  "out": Object {},
  "parents": Object {},
  "ports": Object {
    "id1": Object {
      "data": Object {
        "type": "string",
      },
      "graphicalAttributes": Object {
        "attr": "attr",
      },
      "id": "id1",
      "nodeId": undefined,
      "portType": undefined,
      "position": Object {
        "x": 10,
        "y": 10,
      },
    },
    "id2": Object {
      "data": Object {},
      "graphicalAttributes": Object {
        "type": undefined,
      },
      "id": "id2",
      "nodeId": undefined,
      "portType": undefined,
      "position": Object {
        "x": 10,
        "y": 10,
      },
    },
    "id3": Object {
      "data": Object {},
      "graphicalAttributes": Object {
        "type": undefined,
      },
      "id": "id3",
      "nodeId": undefined,
      "portType": undefined,
      "position": Object {
        "x": 10,
        "y": 10,
      },
    },
  },
  "transform": Object {
    "k": 1,
    "x": 0,
    "y": 0,
  },
  "transformToApply": undefined,
}
`;

exports[`Check port reducer FLOWDESIGNER_PORT_SET_GRAPHICAL_ATTRIBUTES to merge { selected: true } on port id1 graphicalAttribute map 1`] = `
Object {
  "childrens": Object {},
  "in": Object {},
  "links": Object {},
  "nodeTypes": Object {},
  "nodes": Object {
    "nodeId": Object {},
  },
  "out": Object {},
  "parents": Object {},
  "ports": Object {
    "id1": Object {
      "data": Object {
        "type": "test",
      },
      "graphicalAttributes": Object {
        "attr": "attr",
        "selected": true,
      },
      "id": "id1",
      "nodeId": undefined,
      "portType": undefined,
      "position": Object {
        "x": 10,
        "y": 10,
      },
    },
    "id2": Object {
      "data": Object {},
      "graphicalAttributes": Object {
        "type": undefined,
      },
      "id": "id2",
      "nodeId": undefined,
      "portType": undefined,
      "position": Object {
        "x": 10,
        "y": 10,
      },
    },
    "id3": Object {
      "data": Object {},
      "graphicalAttributes": Object {
        "type": undefined,
      },
      "id": "id3",
      "nodeId": undefined,
      "portType": undefined,
      "position": Object {
        "x": 10,
        "y": 10,
      },
    },
  },
  "transform": Object {
    "k": 1,
    "x": 0,
    "y": 0,
  },
  "transformToApply": undefined,
}
`;
